sycl/source/detail/config.hpp

```diff
@@ -579,6 +579,43 @@ template <> class SYCLConfig<SYCL_REDUCTION_PREFERRED_WORKGROUP_SIZE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_ENABLE_FUSION_CACHING> {
+  using BaseT = SYCLConfigBase<SYCL_ENABLE_FUSION_CACHING>;
+
+public:
+  static bool get() {
+    constexpr bool DefaultValue = true;
+
+    const char *ValStr = getCachedValue();
+
+    if (!ValStr)
+      return DefaultValue;
+
+    int Result = DefaultValue;
+    try {
+      Result = std::stoi(ValStr);
+    } catch (...) {
+      throw invalid_parameter_error(
+          "Invalid value for SYCL_ENABLE_FUSION_CACHING environment variable: "
+          "value should be '0' or '1'",
+          PI_ERROR_INVALID_VALUE);
+    }
+    return Result;
```

Comment:
Nit; technically we are allowing any integer value here, so should it check that it's one of the valid ones or maybe just say "Disable ('0') or enable (any non-zero integer value) ..." in the docs?

Reply:
That's true, I adopted what is done for `SYCL_ENABLE_DEFAULT_CONTEXTS` to allow only `1` and `0` values.